### PR TITLE
Add Principles list

### PR DESCRIPTION
**Pull request by vtbassmatt**
Would love some public feedback on this list.

### CODE REVIEW

**Comment by alexmullans** [comment]
I'm not "public feedback" 😛 but this looks great.

File: design/yaml-principles.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Give a way to bail out if we guess wrong.
+3. Follow the principle of least surprise.
+If you have to make a decision with no clear "right" answer, consider which way will be less surprising once learned.
+That'll be easier to *remember*.
+4. Understand what others chose to do and why.
+Our system is different in large and small ways from other CI/CD systems.
+This principle *doesn't* mean "just do what <X> does".
+5. Our language is terse, moreso for common things.
+It's "job", not "jobToRun".
+A well-placed adjective can clear up a lot of confusion, though: it's "displayName", not "name".
```
Comment (ericsciple) [inline design/yaml-principles.md]: One of the things that bites customers, is too many ways of doing the same thing. It however runs counter to the keep-simple-things-simple principle (we should not go overboard).

File: design/yaml-principles.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+Don't forget the deep expert who has to read and write these things day in and day out.
+2. Be opinionated.
+Try to do the right thing automatically.
+Give a way to bail out if we guess wrong.
+3. Follow the principle of least surprise.
+If you have to make a decision with no clear "right" answer, consider which way will be less surprising once learned.
+That'll be easier to *remember*.
+4. Understand what others chose to do and why.
+Our system is different in large and small ways from other CI/CD systems.
+This principle *doesn't* mean "just do what <X> does".
```
Comment (ericsciple) [inline design/yaml-principles.md]:
```md
\"just do what \<X\> does\"
```

File: design/yaml-principles.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# YAML language principles
+
+Here are the principles we're using to evaluate YAML suggestions, especially when they affect the base syntax.
+These are *very* rough in priority order.
+
+1. Make simple things simple, complex things possible.
```
Comment (ericsciple) [inline design/yaml-principles.md]: examples we used for easy, medium, complex scenarios were: single-job (should be simple to describe in yaml), multi-config (medium), and job graph (complex).

File: design/yaml-principles.md (hunk 1 above)
Comment (vtbassmatt) [reply design/yaml-principles.md]: Ooh that's a good one. Let me think about how to work that in.

File: design/yaml-principles.md (hunk 2 above)
Comment (vtbassmatt) [reply design/yaml-principles.md]: Proper escaping and display is so overrated :-P

**Pull request merged**